File: look.py
# def look_disk_scheduling(requests, start_direction="left"): 
#     seek_count = 0
#     current_track = 0
    
#     if start_direction == "left":
#         requests.sort()
#     else:
#         requests.sort(reverse=True)
        
#     for request in requests:
#         seek_count += abs(current_track - request)
#         current_track = request
    
#     return seek_count

# # Example Usage
# requests = [98, 183, 37, 122, 14, 124, 65, 67]
# look_result = look_disk_scheduling(requests, start_direction="left")
# print(f"Look Disk Scheduling Seek Count: {look_result}")


def LOOK(arr, head, direction):
    total_distance = 0
    size = len(arr)

    visited = [0] * size

    # Sorting disk locations
    arr.sort()

    # Finding the position of the head in sorted array
    for i in range(size):
        if arr[i] > head:
            left = i - 1
            right = i
            break

    # LOOK in the left direction
    for i in range(left, -1, -1):
        current = arr[i]
        visited[i] = 1
        distance = abs(current - head)
        total_distance += distance
        head = current

    # LOOK in the right direction
    for i in range(right, size):
        current = arr[i]
        visited[i] = 1
        distance = abs(current - head)
        total_distance += distance
        head = current

    print("Seek Sequence:", arr)
    print("Total Head Movement:", total_distance)


if __name__ == "__main__":
    n = int(input("Enter the number of disk requests: "))
    arr = list(map(int, input("Enter the disk requests: ").split()))
    head = int(input("Enter the initial position of the disk head: "))
    direction = input("Enter the direction (L/R): ")

    LOOK(arr, head, direction)

File: look.c
#include <stdio.h>
#include <stdlib.h>

void LOOK(int arr[], int head, char direction, int size) {
    int distance = 0;
    int current, previous = head;  // Initialize 'previous' with the initial head position
    int i;
    int total_distance = 0;

    int left = 0, right = 0;

    // Sorting
    for (i = 0; i < size - 1; i++) {
        for (int j = i + 1; j < size; j++) {
            if (arr[i] > arr[j]) {
                int temp = arr[i];
                arr[i] = arr[j];
                arr[j] = temp;
            }
        }
    }

    // Head position
    for (i = 0; i < size; i++) {
        if (arr[i] > head) {
            left = i - 1;
            right = i;
            break;
        }
    }

    // Left
    for (i = left; i >= 0; i--) {
        current = arr[i];
        distance = abs(current - previous);
        total_distance += distance;
        previous = current;
    }

    // Right
    for (i = right; i < size; i++) {
        current = arr[i];
        distance = abs(current - previous);
        total_distance += distance;
        previous = current;
    }

    printf("\nSeek Sequence: ");
    for (i = 0; i < size; i++) {
        printf("%d ", arr[i]);
    }

    printf("\nTotal Head Movement: %d\n", total_distance);
}

int main() {
    int head, n, i;
    char direction;

    printf("\nEnter the number of disk requests: ");
    scanf("%d", &n);

    int *arr = (int *)malloc(n * sizeof(int));
    printf("\nEnter the disk requests: ");
    for (i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    printf("\nEnter the initial position of head: ");
    scanf("%d", &head);

    printf("\nEnter the direction(L/R): ");
    scanf(" %c", &direction);

    LOOK(arr, head, direction, n);

    free(arr);
    return 0;
}
